### PR TITLE
Incl. Authorization Token to New Relic Dashboard Plugin

### PR DESCRIPTION
**Pull request by skgandikota**
Signed-off-by: skgandikota <gsaikoushik@yahoo.com>

## Hey, I just made a Pull Request!

Authorising the new relic proxy call from frontend to backend by including  User Authorization token to New Relic Dashboards Plugin.  Appended myHeaders const with Authorization Token [at line 44 ](https://github.com/backstage/backstage/blob/da0675bf9f28ed1460f03635a22d3c26abd14707/plugins/newrelic-dashboard/src/api/NewRelicDashboardClient.ts#L44)from IdentityApi  + new import cascading changes  

#### :heavy_check_mark: Checklist

- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by skgandikota** [comment]
considering the @freben 's comment over issue https://github.com/backstage/backstage/issues/9889 and updating the code  with  fetchRefApi 
https://github.com/backstage/backstage/issues/9889#issuecomment-1095006004

File: .changeset/pink-mayflies-rhyme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-newrelic-dashboard': minor
```
Comment (benjdlambert) [inline .changeset/pink-mayflies-rhyme.md:2]: Think this should be a `patch` change.  As there's no breaking change and the package is `<1.0.0`

File: plugins/newrelic-dashboard/src/api/NewRelicDashboardClient.ts [hunk 2]
```diff
@@ -27,21 +27,27 @@ import { ResponseError } from '@backstage/errors';
 
 export class NewRelicDashboardClient implements NewRelicDashboardApi {
   private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
```
Comment (benjdlambert) [inline plugins/newrelic-dashboard/src/api/NewRelicDashboardClient.ts:30]: Think that it actually makes sense to use the `fetchApi` here, which has the ability to automatically inject the headers necessary from the `identityApi` for you, so you won't need to do anything else.

It should just replace the `fetch` call, and I think looking at the code it should be a drop in replacement.

**Review by benjdlambert — COMMENTED** [review]
Thanks for this! Just a few nits!

File: .changeset/pink-mayflies-rhyme.md (hunk 1 above)
Comment (skgandikota) [reply .changeset/pink-mayflies-rhyme.md]: modified to patch

File: plugins/newrelic-dashboard/src/api/NewRelicDashboardClient.ts (hunk 2 above)
Comment (skgandikota) [reply plugins/newrelic-dashboard/src/api/NewRelicDashboardClient.ts]: Yep ! That's a fantastic suggestion  Thanks  a lot @benjdlambert   I have just added `fetchApi` and committed code https://github.com/backstage/backstage/pull/10754/commits/7eee3434aacd2a72d1ac5823b3f23a44db92f76a  , please have a look and let me know for any changes

**Comment by skgandikota** [comment]
duplicate request - closing

**Pull request closed without merging**